lab-said/src/main.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import ReactDom from 'react-dom';
+import superagent from 'superagent';
+
+const API_URL = 'http://www.reddit.com/r';
+
+let renderIf = (test, component) => test ? component : undefined;
+
+class RedditForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      topicSearch: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    console.log('__FORM_PROPS__', this.props);
+    console.log('__FORM_STATE__', this.state);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.props.topicSelect(this.state.topicSearch);
+  }
+
+  handleChange(e) {
+    this.setState({topicSearch: e.target.value});
+  }
+
+  render() {
+    return (
+      <form
+        className="topics-form"
+        onSubmit={this.handleSubmit}>
+
+        <input
```

Comment:
there also needs to be a place to input a number, and limit the number of results you get back from the call.